### PR TITLE
samples: nrf9160: aws_iot: Add printing of received data

### PR DESCRIPTION
**Pull request by simensrostad**
This patch adds functionality that prints received
messages and what topic the messages was received on.

NCSDK-6623

### CODE REVIEW

File: samples/nrf9160/aws_iot/src/main.c [hunk 1]
```diff
@@ -176,6 +176,33 @@ static void shadow_update_version_work_fn(struct k_work *work)
...
+	str = cJSON_Print(root_obj);
+	if (str == NULL) {
+		printk("Failed to print JSON object");
+		goto clean_exit;
+	}
+
+	printf("Data received from AWS IoT console:\nTopic: %.*s\nMessage: %s\n",
+	       topic_len, topic, str);
+
+	k_free(str);
```
Comment (rlubos) [inline samples/nrf9160/aws_iot/src/main.c:200]: nit; There's a `cJSON_FreeString` helper in the `cJSON_os.h` header, would be good to use it instead of direct `k_free()` call.

Comment (simensrostad) [reply samples/nrf9160/aws_iot/src/main.c]: Changed for both occurrences in main.c

**Review by rlubos — APPROVED** [review]

**Comment by pkchan** [comment]
I have verified functionality from AWS cloud. It's working fine.

> RRC mode: Connected
AWS_IOT_EVT_DATA_RECEIVED
Data received from AWS IoT console:
> 
> Topic: my-custom-topic/example
> 
> Message: {
> 
> 	"message":	"Test 1"
> 
> }
> 
> RRC mode: Idle

**Review by pkchan — APPROVED** [review]

File: samples/nrf9160/aws_iot/src/main.c [hunk 2]
```diff
@@ -141,7 +141,7 @@ static int shadow_update(bool version_number_include)
 		printk("aws_iot_send, error: %d\n", err);
 	}
 
-	k_free(message);
+	cJSON_FreeString(message);
```
Comment (lemrey) [inline samples/nrf9160/aws_iot/src/main.c:144]: Should this be a separate commit?

Comment (simensrostad) [reply samples/nrf9160/aws_iot/src/main.c:144]: Done

**Review by lemrey — APPROVED** [review]
Thank you

**Review by shantha-14 — APPROVED** [review]
